src/robot/libraries/String.py

```diff
@@ -237,29 +237,38 @@ def get_lines_matching_pattern(self, string, pattern, case_insensitive=False):
             matches = lambda line: fnmatchcase(line, pattern)
         return self._get_matching_lines(string, matches)
 
-    def get_lines_matching_regexp(self, string, pattern):
+    def get_lines_matching_regexp(self, string, pattern, exact_match=True):
         """Returns lines of the given ``string`` that match the regexp ``pattern``.
 
         See `BuiltIn.Should Match Regexp` for more information about
         Python regular expression syntax in general and how to use it
-        in Robot Framework test data in particular. A line matches
-        only if it matches the ``pattern`` fully. Notice that to make
-        the match case-insensitive, you need to embed case-insensitive
-        flag into the pattern.
-
-        Lines are returned as one string catenated back together with
+        in Robot Framework test data in particular. By default ``exact_match``
+        is True causing line matches only if it matches the ``pattern`` fully.
+        When passing ``exact_match`` as False, this keyword uses the matching
+        pattern without the '^' and '$' positional markers, searching content
+        within the input lines. Notice that to make the match case-insensitive,
+        you need to embed case-insensitive flag into the pattern.
+
+        Lines are returned as one string concatenated back together with
         newlines. Possible trailing newline is never returned. The
         number of matching lines is automatically logged.
 
         Examples:
-        | ${lines} = | Get Lines Matching Regexp | ${result} | Reg\\\\w{3} example |
-        | ${ret} = | Get Lines Matching Regexp | ${ret} | (?i)FAIL: .* |
-
-        See `Get Lines Matching Pattern` and `Get Lines Containing String` if
-        you do not need full regular expression powers (and complexity).
+        | ${lines} = | Get Lines Matching Regexp | ${result} | Reg\\\\w{3} example |   |
+        | ${ret} = | Get Lines Matching Regexp | ${ret} | (?i)FAIL: .* |   |
+        | ${result} = | Get Lines Matching Regexp | ${input} | ^ABC | False |
+        | ${result} = | Get Lines Matching Regexp | ${input} | ^ABC.* | True |
```

Comment:
Good that you've added examples. I'd like them to be enhanced slightly:
- Last two examples should be in different order with the one using default behavior first.
- Using `exact_match=False` would be much more explicit than just `False`. There's no need to have `True` at all because it is the default.
- Both should have same pattern to make it explicit that the value of `exact_match` is interesting. Using `^` in the pattern requiring exact match is confusing.